lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -742,56 +742,28 @@ void OCLToSPIRVBase::visitCallConvert(CallInst *CI, StringRef MangledName,
   if (auto *VecTy = dyn_cast<VectorType>(SrcTy))
     SrcTy = VecTy->getElementType();
   auto IsTargetInt = isa<IntegerType>(TargetTy);
-  auto TargetSigned = DemangledName[8] != 'u';
 
-  std::string TargetTyName(
+  std::string ConversionFunc(
       DemangledName.substr(strlen(kOCLBuiltinName::ConvertPrefix)));
-  auto FirstUnderscoreLoc = TargetTyName.find('_');
-  if (FirstUnderscoreLoc != std::string::npos)
-    TargetTyName = TargetTyName.substr(0, FirstUnderscoreLoc);
 
-  // Validate target type name
-  std::regex Expr("([a-z]+)([0-9]*)$");
+  // Validate target type name and vector size if present
+  std::regex Expr("(float|double|half|u?char|u?short|u?int|u?long)(2|3|4|8|16)*"
```

Comment:
Is it possible to have the regex start with kOCLBuiltinName::ConvertPrefix so that we don't have to use DemangedName.substr?

Reply:
Applied